src/devices/Board/PinUsage.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Iot.Device.Board
+{
+    /// <summary>
+    /// Designated (or active) usage of a pin
+    /// </summary>
+    public enum PinUsage
+    {
+        /// <summary>
+        /// Pin not currently used (or usage unknown)
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// Pin used for GPIO (input or output)
+        /// </summary>
+        Gpio = 1,
+
+        /// <summary>
+        /// Pin used for I2C
+        /// </summary>
+        I2c = 2,
+
+        /// <summary>
+        /// Pin used for SPI
+        /// </summary>
+        Spi = 3,
+
+        /// <summary>
+        /// Pin used for PWM (or analog out)
+        /// </summary>
+        Pwm = 4,
+
+        /// <summary>
+        /// Pin used for RS-232
+        /// </summary>
+        Uart = 5,
+
+        /// <summary>
+        /// Pin used for analog input
+        /// </summary>
+        AnalogIn = 6,
```

Comment:
Shall we add an AnalogOut as well?

Reply:
Is an option, but I'm not sure... Many devices I know use the PWM interface to also control those.

Reply:
GrovePi offers real Analog Output for example. It will then perfectly fit here.

Reply:
I know, some arduinos also have real DACs. What I meant is that some boards use the same interface methods to set a value to a DAC like to set a PWM pin (using just the duty cycle and ignoring the frequency). But that should probably stay implementation-dependent. I'll add the value.

Reply:
Pi-top also have ping used for analog